src/prefect/runner/server.py

```diff
@@ -0,0 +1,79 @@
+import pendulum
+import uvicorn
+from prefect._vendor.fastapi import APIRouter, FastAPI, status
+from prefect._vendor.fastapi.responses import JSONResponse
+
+from prefect.settings import (
+    PREFECT_RUNNER_POLL_FREQUENCY,
+    PREFECT_RUNNER_SERVER_HOST,
+    PREFECT_RUNNER_SERVER_LOG_LEVEL,
+    PREFECT_RUNNER_SERVER_MISSED_POLLS_TOLERANCE,
+    PREFECT_RUNNER_SERVER_PORT,
+)
+
+
+def perform_health_check(runner, delay_threshold: int = None) -> JSONResponse:
+    if delay_threshold is None:
+        delay_threshold = (
+            PREFECT_RUNNER_SERVER_MISSED_POLLS_TOLERANCE.value()
+            * PREFECT_RUNNER_POLL_FREQUENCY.value()
+        )
+
+    def _health_check():
+        now = pendulum.now("utc")
+        poll_delay = (now - runner.last_polled).total_seconds()
+
+        if poll_delay > delay_threshold:
+            return JSONResponse(
+                status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+                content={"message": "Runner is unresponsive at this time"},
+            )
+        return JSONResponse(status_code=status.HTTP_200_OK, content={"message": "OK"})
+
+    return _health_check
+
+
+def run_count(runner) -> int:
+    def _run_count():
+        run_count = len(runner._flow_run_process_map)
+        return run_count
+
+    return _run_count
+
+
+def shutdown(runner) -> int:
+    def _shutdown():
+        runner.should_stop = True
```

Comment:
I'd vote to move this into `Runner.stop()` to ensure the polling loops stop anytime `.stop` is called.